### PR TITLE
Dsa assignment8

### PR DESCRIPTION
**Pull request by meta-anshita-arya**
submitted

### CODE REVIEW

File: PF/Assignment1/TreeSort/TreeSortImplementation<T>.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+//class to perform binary search tree operations
+public class TreeSortImplementation<T> {
+	int index = 0;
```
Comment (certinia2023) [inline PF/Assignment1/TreeSort/TreeSortImplementation<T>.java:3]: ACTION: move this variable declaration inside inorder().

File: PF/Assignment1/TreeSort/TreeSortMain.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		TreeSortImplementation<Integer> treeSort = new TreeSortImplementation<Integer>();
+		int noOfElements;
+		// Input number of elements from user
+		System.out.println("Enter no. of elements");
+		noOfElements = scan.nextInt();
+		int arr[] = new int[noOfElements];
+		int result[] = new int[noOfElements];
+		// Input roll numbers from user
+		System.out.println("Enter all the roll numbers of students");
+		arr[0] = scan.nextInt();
```
Comment (certinia2023) [inline PF/Assignment1/TreeSort/TreeSortMain.java:18]: ACTION: Why you take single input here?

File: PF/Assignment1/TreeSort/TreeSortMain.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		Node<Integer> node = new Node<Integer>(arr[0]);
```
Comment (certinia2023) [inline PF/Assignment1/TreeSort/TreeSortMain.java:19]: ACTION: There is no need to initialize node here with first element as you already have a null check in insertTree.

File: PF/Assignment1/TreeSort/TreeSortTest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	}
+
+	// Positive Test Case
+	@Test
+	public void testForPositive() {
+		int arr[] = { 1, 5, 2, 4, 3 };
+		int result[] = { 1, 2, 3, 4, 5 };
+		int output[] = new int[5];
+		TreeSortImplementation<Integer> treeSort = new TreeSortImplementation<Integer>();
+		Node<Integer> node = new Node<Integer>(arr[0]);
```
Comment (certinia2023) [inline PF/Assignment1/TreeSort/TreeSortTest.java:38]: ACTION: Use unparameterized constructor to initialize Node.

File: PF/Assignment1/TreeSort/TreeSortTest.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	public void setUp() throws Exception {
+	}
+
+	@After
+	public void tearDown() throws Exception {
+	}
+
+	// Positive Test Case
+	@Test
+	public void testForPositive() {
```
Comment (certinia2023) [inline PF/Assignment1/TreeSort/TreeSortTest.java:33]: ACTION: Use proper test method name.

File: PF/Assignment1/TreeSort/TreeSortTest.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+		for (int i = 1; i < 5; i++) {
+			treeSort.insertTree(node, arr[i]);
+		}
+		output = treeSort.inorder(node, arr);
+		assertArrayEquals(output, result);
+	}
+
+	// Negative Test Case
+	@Test
+	public void testForNegative() {
```
Comment (certinia2023) [inline PF/Assignment1/TreeSort/TreeSortTest.java:48]: ACTION: Use proper test method name.

File: PF/Assignment1/TreeSort/TreeSortTest.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+		int output[] = new int[5];
+		TreeSortImplementation<Integer> treeSort = new TreeSortImplementation<Integer>();
+		Node<Integer> node = new Node<Integer>(arr[0]);
+		for (int i = 1; i < 5; i++) {
+			treeSort.insertTree(node, arr[i]);
+		}
+		output = treeSort.inorder(node, arr);
+		assertNotEquals(result, output);
+	}
+
```
Comment (certinia2023) [inline PF/Assignment1/TreeSort/TreeSortTest.java:60]: ACTION: Add test for duplicate numbers.

File: PF/Assignment1/TreeSort/TreeSortMain.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+		for (int i = 1; i < noOfElements; i++) {
+			int checkDuplicate = 0;
+			while (checkDuplicate == 0) {
+				arr[i] = scan.nextInt();
+				int count = 0;
```
Comment (certinia2023) [inline PF/Assignment1/TreeSort/TreeSortMain.java:24]: ACTION: Create a method to check duplicate value in array and also cover that in test method.

File: PF/Assignment1/TreeSort/TreeSortTest.java [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+	// Negative Test Case
+	@Test
+	public void testForNegative() {
+		int arr[] = { 1, 5, 2, 4, 3 };
+		int result[] = { 0, 3, 2, 4, 5 };
+		int output[] = new int[5];
+		TreeSortImplementation<Integer> treeSort = new TreeSortImplementation<Integer>();
+		Node<Integer> node = new Node<Integer>(arr[0]);
```
Comment (certinia2023) [inline PF/Assignment1/TreeSort/TreeSortTest.java:53]: ACTION: Use unparameterized constructor to initialize Node.

File: PF/Assignment1/TreeSort/TreeSortTest.java [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+		for (int i = 1; i < 5; i++) {
+			treeSort.insertTree(node, arr[i]);
+		}
+		output = treeSort.inorder(node, arr);
+		assertNotEquals(result, output);
+	}
+
+	// Test Case for no node value
+	@Test
+	public void testForNull() {
```
Comment (certinia2023) [inline PF/Assignment1/TreeSort/TreeSortTest.java:63]: ACTION: Use proper name for test method.

File: PF/Assignment1/Sort/Sort.java [hunk 11]
```diff
@@ -0,0 +1,44 @@
+//main class
+public class Sort {
+
+	// main function
+	public static void main(String arg[]) {
+		MenuAndPrint menuandprint = new MenuAndPrint();
+		int[] array = menuandprint.arrayInput();
+		int length = array.length;
```
Comment (certinia2023) [inline PF/Assignment1/Sort/Sort.java:8]: ACTION: no need to create a new variable to store length of array use directly array.length.

File: PF/Assignment1/Sort/BubbleSort.java [hunk 12]
```diff
@@ -0,0 +1,20 @@
+public class BubbleSort {
+
+	// for bubble sort
+	int[] bubbleSort(int[] array, int length) {
+		int forSwap = 0;
```
Comment (certinia2023) [inline PF/Assignment1/Sort/BubbleSort.java:5]: ACTION: Use proper variable name.

File: PF/Assignment1/Sort/RadixSort.java [hunk 13]
```diff
@@ -0,0 +1,40 @@
+public class RadixSort {
+
+	// for radix sort
+	int[] radixSort(int[] array, int length, int noOfDigit) {
+
+		int forQuotient = 0;
```
Comment (certinia2023) [inline PF/Assignment1/Sort/RadixSort.java:6]: ACTION: Use proper name for variable.
